common/src/main/java/org/opensearch/ml/common/conversational/ConversationMeta.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.ml.common.conversational;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.search.SearchHit;
+
+/**
+ * Class for holding conversational metadata
+ */
+public final class ConversationMeta implements Writeable, ToXContentObject {
+
+    private String id;
+    private Instant created;
+    private Instant lastHit;
+    private int numInteractions;
+    private String name;
+    private String user;
+
+    /**
+     * Most naive constructor
+     * @param id the UID of this conversation
+     * @param created the timestamp of this conversation's creation
+     * @param lastHit the timestamp of the most recent interaction belonging to this conversation 
+     *      or the time of creation if there are no interactions
+     * @param numInteractions the length of this conversation
+     * @param name a user-defined name for the conversation
+     * @param user the user who owns this conversation
+     */
+    public ConversationMeta(
```

Comment:
Can we keep the same style comparing with another pr https://github.com/opensearch-project/ml-commons/pull/1195/files#diff-b03c98561891e1281b5d5851ee7aad5d4589651158dead6fe9bdbdcf7072ff5fR27? Let's use lombok here.